vyper/codegen/core.py

```diff
@@ -290,6 +301,48 @@ def append_dyn_array(darray_node, elem_node):
             return IRnode.from_list(b1.resolve(b2.resolve(ret)))
 
 
+def extend_dyn_array(context, dst, src):
+    assert isinstance(dst.typ, DArrayType)
+    assert isinstance(src.typ, DArrayType)
+
+    ret = ["seq"]
+
+    with dst.cache_when_complex("darray_dst") as (b1, dst):
+        dst_len = get_dyn_array_count(dst)
+        src_len = get_dyn_array_count(src)
+
+        with dst_len.cache_when_complex("dst_darray_len") as (
+            b2,
+            dst_len,
+        ), src_len.cache_when_complex("src_darray_len") as (b3, src_len):
+
+            max_dst_len = dst.typ.count
+            combined_len = IRnode.from_list(["add", dst_len, src_len], typ="uint256")
+
+            # Assert that `src_len + dst_len` <= maxlen(dst)`
+            within_maxlen_assertion = IRnode.from_list(
```

Comment:
nit: i would rename this to `bounds_check`, or even simply `check` (which is the convention in the safemath routines)